### PR TITLE
Check all env vars and suggest alternatives for misspelled keys

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

This PR adds checking for all `FTLCONF_*` environment variables. We log used ones, invalid ones (e.g. `123` for something expecting a boolean), and unknown keys (ignored). In the latter case, we try to suggest a meaningful alternative to aid typo fixing.

Example environment:
```
FTLCONF_misc_nice=-12
FTLCONF_misc_privacyllll=-12
FTLCONF_dns_prt=123
FTLCONF_debug_api=not_a_bool
FTLCONF_dns_upstrrrr=1,2,3
FTLCONF_debug_ApI=true
```
leading to the following log output:
(removed, see updated output in comment below)

When no environment variables are defined, nothing environmental variables-related is logged at all.

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: src/config/env.c [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+		{
+			if(item->valid)
+				log_info("%s %s is used", cli_tick(), item->key);
+			else
+				log_info("%s %s is invalid, using default", cli_cross(), item->key);
+			continue;
+		}
+		// else: print warning
+		const char *closest_match = suggest_closest(env_keys, sizeof(env_keys) / sizeof(*env_keys), item->key);
+		log_info("%s %s is unknown and ignored, did you mean %s ?", cli_qst(), item->key, closest_match);
```
Comment (yubiuser) [inline src/config/env.c:105]: There are several places where warnings should be logged, but `info` is done.

**Review by yubiuser — COMMENTED** [review]
```
FTLCONF_debug_api is invalid, using default

```
Would be nice if the log would tell us, why it is invalid. E.g.

```
FTLCONF_debug_api is invalid (expecting bool, string set), using default
```

File: src/config/env.c (hunk 1 above)
Comment (DL6ER) [reply src/config/env.c:105]: I did this intentionally so the text is more readable, compare:
```
2023-10-31 19:20:03.366 [799762M] INFO: 6 FTLCONF environment variables found (1 used, 1 invalid, 4 ignored)
2023-10-31 19:20:03.367 [799762M] INFO: [?] FTLCONF_debug_ApI is unknown and ignored, did you mean FTLCONF_debug_api ?
2023-10-31 19:20:03.367 [799762M] INFO: [?] FTLCONF_dns_upstrrrr is unknown and ignored, did you mean FTLCONF_dns_upstreams ?
2023-10-31 19:20:03.367 [799762M] INFO: [✗] FTLCONF_debug_api is invalid, using default
2023-10-31 19:20:03.368 [799762M] INFO: [?] FTLCONF_dns_prt is unknown and ignored, did you mean FTLCONF_dns_port ?
2023-10-31 19:20:03.368 [799762M] INFO: [?] FTLCONF_misc_privacyllll is unknown and ignored, did you mean FTLCONF_misc_privacylevel ?
2023-10-31 19:20:03.368 [799762M] INFO: [✓] FTLCONF_misc_nice is used
```
vs.
```
2023-10-31 19:20:03.366 [799762M] INFO: 6 FTLCONF environment variables found (1 used, 1 invalid, 4 ignored)
2023-10-31 19:20:03.367 [799762M] WARNING: [?] FTLCONF_debug_ApI is unknown and ignored, did you mean FTLCONF_debug_api ?
2023-10-31 19:20:03.367 [799762M] WARNING: [?] FTLCONF_dns_upstrrrr is unknown and ignored, did you mean FTLCONF_dns_upstreams ?
2023-10-31 19:20:03.367 [799762M] ERROR: [✗] FTLCONF_debug_api is invalid, using default
2023-10-31 19:20:03.368 [799762M] WARNING: [?] FTLCONF_dns_prt is unknown and ignored, did you mean FTLCONF_dns_port ?
2023-10-31 19:20:03.368 [799762M] WARNING: [?] FTLCONF_misc_privacyllll is unknown and ignored, did you mean FTLCONF_misc_privacylevel ?
2023-10-31 19:20:03.368 [799762M] INFO: [✓] FTLCONF_misc_nice is used
```

We could space-align to make it

vs.
```
2023-10-31 19:20:03.366 [799762M] INFO:     6 FTLCONF environment variables found (1 used, 1 invalid, 4 ignored)
2023-10-31 19:20:03.367 [799762M] WARNING: [?] FTLCONF_debug_ApI is unknown and ignored, did you mean FTLCONF_debug_api ?
2023-10-31 19:20:03.367 [799762M] WARNING: [?] FTLCONF_dns_upstrrrr is unknown and ignored, did you mean FTLCONF_dns_upstreams ?
2023-10-31 19:20:03.367 [799762M] ERROR:   [✗] FTLCONF_debug_api is invalid, using default
2023-10-31 19:20:03.368 [799762M] WARNING: [?] FTLCONF_dns_prt is unknown and ignored, did you mean FTLCONF_dns_port ?
2023-10-31 19:20:03.368 [799762M] WARNING: [?] FTLCONF_misc_privacyllll is unknown and ignored, did you mean FTLCONF_misc_privacylevel ?
2023-10-31 19:20:03.368 [799762M] INFO:    [✓] FTLCONF_misc_nice is used
```

**Comment by DL6ER** [comment]
Rebased on latest `development-v6`

**Comment by DL6ER** [comment]
@yubiuser Sorry, I did not explicitly reply to your question. What you'd like to see was hidden in my example above:
```
2023-11-01 06:52:11.637 [1067602M] ERROR:   [✗] FTLCONF_debug_api is invalid (not of type bool), using default
```

I did not include what was set instead as *all* environmental variables are always strings. Guessing what the string contains is ambiguous and settings should only be interpreted how they are defined. For instance, `-11` makes sense for `misc.nice` but not for `dns.port` even when both accept numbers.

**Review by PromoFaux — APPROVED** [review]
```
pihole  | 2024-02-10 08:25:00.174 [146M] WARNING: [?] FTLCONF_webserver_aip_passwordddd is unknown, did you mean any of these?
pihole  | 2024-02-10 08:25:00.174 [146M] WARNING:     - FTLCONF_webserver_api_password
```

**Pull request merged**